### PR TITLE
add: structure inception v4

### PR DESCRIPTION
**Pull request by makaveli10**

### CODE REVIEW

File: inceptionv4/engine.h [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+#include "NvInfer.h"
+#include "cuda_runtime_api.h"
+
+static Logger gLogger;
+
+namespace trtx {
+    struct InferenceEngineConfig {
+        const char* input_blob_name;
+        const char* output_blob_name; 
+        int max_batch_size;
```
Comment (shivamswarnkar) [inline inceptionv4/engine.h:18]: Any reasons to not use CamelCase notation here? (for line 16, 17, 18)

File: inceptionv4/utils.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+#define CHECK(status)                             \
+    do                                            \
+    {                                             \
+        auto ret = (status);                      \
+        if (ret != 0)                             \
+        {                                         \
+            std::cout << "Cuda failure: " << ret; \
+            abort();                              \
+        }                                         \
+    } while (0)
```
Comment (shivamswarnkar) [inline inceptionv4/utils.h:23]: what's the point of this do-while loop? It'll always run once, right? Am I missing something?

File: inceptionv4/logging.h [hunk 3]
```diff
@@ -0,0 +1,507 @@
...
+#include <iomanip>
+#include <iostream>
+#include <ostream>
+#include <sstream>
+#include <string>
+
+using Severity = nvinfer1::ILogger::Severity;
+
+class LogStreamConsumerBuffer : public std::stringbuf
+{
```
Comment (shivamswarnkar) [inline inceptionv4/logging.h:32]: Little formatting inconsistency. In this file, every opening parenthesis (for function, class def etc.) is placed in the next line, while in other files they are placed on the same line.

**Review by shivamswarnkar — APPROVED** [review]

File: inceptionv4/logging.h (hunk 3 above)
Comment (makaveli10) [reply inceptionv4/logging.h:32]: ye nvidia ka file h. vahi utha liya

File: inceptionv4/engine.h (hunk 1 above)
Comment (makaveli10) [reply inceptionv4/engine.h:18]: No it should be camel case. missed this I guess

File: inceptionv4/utils.h (hunk 2 above)
Comment (shivamswarnkar) [reply inceptionv4/utils.h:23]: got it, thanks!

File: inceptionv4/logging.h (hunk 3 above)
Comment (shivamswarnkar) [reply inceptionv4/logging.h:32]: ohk.

**Review by aditya-dl — APPROVED** [review]

**Pull request merged**